src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface Goods {
+  sortField: SortType | '';
+  reverse: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortField, reverse }: Goods) {
+  const preparedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.SORT_BY_ALPHABET:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.SORT_BY_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverse,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.SORT_BY_ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.SORT_BY_ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortField(SortType.SORT_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setReverse(!reverse)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== '' || reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReverse(false);
+            }}
+          >
```

Comment:
```suggestion
            onClick={reset}
          >
```
Could you consider moving this function from the JSX? It would allow us to reuse this function and optimize it with the useCallback hook in the future